tools/talis/config.go

```diff
@@ -0,0 +1,246 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync/atomic"
+)
+
+type NodeType string
+
+const (
+	// Validator represents a validator node in the network.
+	Validator NodeType = "validator"
+	// Bridge represents a bridge node in the network.
+	Bridge NodeType = "bridge"
+	// Light represents a light node in the network.
+	Light NodeType = "light"
+)
+
+var (
+	valCount   = atomic.Uint32{}
+	nodeCount  = atomic.Uint32{}
+	lightCount = atomic.Uint32{}
+)
+
+// NodeName returns the name of the node based on its type and index. The
+// name is in the format "<node_type>-<index>". For example, "validator-0" or
+// "bridge-1". Index is a global counter that is incremented for each node created.
+func NodeName(nodeType NodeType) string {
+	index := 0
+	switch nodeType {
+	case Validator:
+		index = int(valCount.Add(1)) - 1
+	case Bridge:
+		index = int(nodeCount.Add(1)) - 1
+	case Light:
+		index = int(lightCount.Add(1)) - 1
+	default:
+		panic(fmt.Sprintf("unknown node type: %s", nodeType))
+	}
+	return fmt.Sprintf("%s-%d", nodeType, index)
+}
+
+// Provider simply marks the provider the instance config should target.
+type Provider string
+
+const (
+	// DO represents DigitalOcean as a provider.
+	DigitalOcean Provider = "digitalocean"
+	// Linode represents Linode as a provider.
+	Linode Provider = "linode"
+)
+
+// Instance represents a single instance in the network. It contains
+// information about the instance such as its public and private IP address,
+// provider, region, and name. It also contains a list of tags that are
+// attached to the instance.
+type Instance struct {
+	NodeType NodeType `json:"node_type"`
+	// PublicIP is the public IP address of the instance.
+	PublicIP string `json:"public_ip"`
+	// PrivateIP is the private IP address of the instance.
+	PrivateIP string `json:"private_ip"`
+	// Provider is the provider of the instance. For example, "digitalocean" or
+	// "aws".
+	Provider Provider `json:"provider"`
+	// Slug is a provider speicific string that determines what type of instance
+	// the node is ran on.
+	Slug string `json:"slug"`
+	// Region is the region in which the instance is created. For example,
+	// "nyc1" for DigitalOcean or "us-east-1" for AWS.
```

Comment:
AWS doesn't seem supported.

```suggestion
	// "nyc1" for DigitalOcean.
```

Reply:
we might not support aws, but this was added on purpose to illustrate that these are different slugs per provider and everyone recognizes use-east-1 and what it means